src/System.Memory/src/System/Buffers/SpanLiteralExtensions.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Buffers
+{
+    internal class SpanLiteralExtensions
+    {
+        internal static void AppendAsLiteral(ReadOnlySpan<byte> span, StringBuilder sb)
+        {
+            for (int i = 0; i < span.Length; i++)
+            {
+                AppendCharLiteral((char) span[i], sb);
```

Comment:
What is the encoding that this assumes?

Casting `(byte)` to `(char)` works for ASCII, but not for anything else. Not very globalization friendly.

Reply:
This is for span of bytes. The ToString change is for span of chars. What this implementation would  be once that change lands?

Reply:
It would reflect `Span<>`: `"ReadOnlyBuffer<Byte>"` for bytes, string representation for `ReadOnlyBuffer<char>`.

Reply:
That means you will lose the textual formatting for array of bytes that you are using it for. If it is the case, I think you can switch to it already and delete this.

Reply:
Yea, I'm not sure we want to remove this. It's pretty useful for debugging, though it's not a common pattern in existing .NET APIs.

Reply:
If we are keeping this, we should think about what encoding is this hard coded to. The current implementation does not match any known encoding. I do not think that it has the right behavior for non-ASCII characters.

Reply:
The current implementation is somewhat useful for text protocols but would be terrible for binary ones, \ux000 notation is very noisy and bulky.